### PR TITLE
Update toctree

### PR DESCRIPTION
**Pull request by ratan**
- update toctree with missing chapter entry

### CODE REVIEW

File: chapters/en/_toctree.yml [hunk 1]
```diff
@@ -13,6 +13,14 @@
       title: What Is Computer Vision
     - local: Unit 1 - Fundamentals/Chapter 1 - Fundamentals of Computer Vision/applications
       title: Applications of Computer Vision
+  - sub-title: feature-extraction
```
Comment (bellabf) [inline chapters/en/_toctree.yml:16]: Feature extraction and image and image and imaging are not in the correct order. It is supposed image and imaging first and then feature extraction

File: chapters/en/_toctree.yml [hunk 2]
```diff
@@ -13,6 +13,14 @@
...
+    sections:
+    - local: Unit 1 - Fundamentals/feature-extraction/feature-matching
+      title: Feature Matching
+    - local: Unit 1 - Fundamentals/feature-extraction/feature_description
+      title: Feature Description
+    - local: Unit 1 - Fundamentals/feature-extraction/real-world-applications
+      title: Real-world Applications of Feature Extraction in Computer Vision
   - sub-title: image_and_imaging
```
Comment (bellabf) [inline chapters/en/_toctree.yml:24]:
```suggestion
  - sub-title: Image and Imaging
```

File: chapters/en/_toctree.yml (hunk 1 above)
Comment (bellabf) [inline chapters/en/_toctree.yml:16]:
```suggestion
  - sub-title: Feature Extraction
```

File: chapters/en/_toctree.yml [hunk 3]
```diff
@@ -24,7 +32,41 @@
...
+      title: Multimodal Text Generation (BLIP)
+    - local: Unit 4 - Multimodal Models/CLIP and relatives/clip
+      title: Contrastive Language-Image Pre-training (CLIP)
+    - local: Unit 4 - Multimodal Models/CLIP and relatives/losses
+      title: Losses
+    - local: Unit 4 - ultimodal Models/CLIP and relatives/owl_vit
+      title: Multimodal Object Detection (OWL-ViT)
   sections:
   - local: Unit 4 - Multimodal Models/pre-intro
     title: What you'll learn and what you'll build
```
Comment (bellabf) [inline chapters/en/_toctree.yml:72]:
```suggestion
    title: What you will learn and what you will build
```

File: chapters/en/_toctree.yml [hunk 4]
```diff
@@ -39,10 +81,80 @@
...
+  sections:
+  - local: Unit 6 - Basic CV Tasks/segmentation
+    title: What is Image Segmentation?
+
+- title: Unit 7. Video and Video Processing
   sections:
-  - local: Unit 5 - Generative Models/Diffusion models
-    title: stable_diffusion
+  - local: Unit 7 - Video and Video Processing/introduction-to-video
+    title: Introduction
```
Comment (bellabf) [inline chapters/en/_toctree.yml:121]:
```suggestion
    title: Introduction to Video
```

File: chapters/en/_toctree.yml [hunk 5]
```diff
@@ -39,10 +81,80 @@
...
+  - local: Unit 7 - Video and Video Processing/video-processing-basics
+    title: Video Processing Basics
+
+- title: Unit 8. 3D Vision, Scene Rendering and Reconstruction
+  - sub-title: 3D Vision
+    sections:
+    - local: Unit 8 - 3D Vision, Scene Rendering and Reconstruction/3D Vision/NVS
```
Comment (bellabf) [inline chapters/en/_toctree.yml:128]: Is this really the first chapter of this unit?

File: chapters/en/_toctree.yml [hunk 6]
```diff
@@ -61,15 +173,31 @@
...
+    title: Hugging Face's efforts - Ethics and Society
   - local: Unit 12 - Ethics and Biases/supplementary-material
     title: Supplementary reading and resources
+
+- title: Unit 13. Outlook
+  sections:
+  - local: Unit 13 - Outlook/Retention - overview
+    title: Retention In Vision
+  - local: Unit 13 - Outlook/hiera
+    title: Overview
```
Comment (bellabf) [inline chapters/en/_toctree.yml:199]:
```suggestion
    title: Overview of Hiera
```

File: chapters/en/_toctree.yml [hunk 7]
```diff
@@ -61,15 +173,31 @@
...
+  - local: Unit 13 - Outlook/hyena
+    title: Overview
```
Comment (bellabf) [inline chapters/en/_toctree.yml:201]:
```suggestion
    title: Overview of Hyena
```

File: chapters/en/_toctree.yml [hunk 8]
```diff
@@ -24,7 +32,41 @@
...
+  - local: Unit 3 - Vision Transformers/Convolutional Vision Transformer
+    title: Convolutional Vision Transformer (CvT)
+  - local: Unit 3 - Vision Transformers/Dinat Transformer
+    title: Dilated Neighborhood Attention Transformer (DINAT)
+  - local: Unit 3 - Vision Transformers/MobileVIT
+    title: MobileViT v2
+  - local: Unit 3 - Vision Transformers/Swin Transformer
+    title: Swin Transformer
+
 - title: Unit 4. Multimodal Models
```
Comment (bellabf) [inline chapters/en/_toctree.yml:57]: I don't think the chapters for unit 4 are in the correct order.

File: chapters/en/_toctree.yml [hunk 9]
```diff
@@ -39,10 +81,80 @@
   - local: Unit 4 - Multimodal Models/supplementary-material
     title: Supplemental reading and resources
  
-- title: Unit 5. Multimodal Models
+- title: Unit 5. Generative Models
```
Comment (bellabf) [inline chapters/en/_toctree.yml:84]: Same thing here, the order is not correct.

File: chapters/en/_toctree.yml [hunk 10]
```diff
@@ -3,7 +3,7 @@
   - local: chapter0/1
     title: Introduction
 
-- title: Unit 1 - Fundamentals
+- title: Unit 1. - Fundamentals
```
Comment (bellabf) [inline chapters/en/_toctree.yml:6]:
```suggestion
- title: Unit 1. Fundamentals
```

**Review by bellabf — COMMENTED** [review]
Hi! For some units, the files are not in the correct order. I left some suggestions for some better naming within certain units.

File: chapters/en/_toctree.yml (hunk 1 above)
Comment (ratan) [reply chapters/en/_toctree.yml]: Done

File: chapters/en/_toctree.yml (hunk 8 above)
Comment (ratan) [reply chapters/en/_toctree.yml:58]: How to find the correct order of chapters for unit 4?

File: chapters/en/_toctree.yml (hunk 9 above)
Comment (ratan) [reply chapters/en/_toctree.yml:85]: How to find the correct order of chapter?

**Comment by ratan** [comment]
Hi @mmhamdy Thanks for the feedback. I restricted this PR only to toc file now.

**Comment by ratan** [comment]
@bellabf  @mmhamdy thanks for your feedback and latest commit has all your recommendations.
I see that many other PR are touching the toc file for few chapters and it will be rebase issue as this PR changes many chapter first time in toc file, making many lines of changes.
Can we merge this PR in this form?
@bellabf i have reached to chapter owner for correct ordering, but only hearf from one whe will revert back by next week. Can i fix the reorder issue in chap 4, 5, 8 in next PR?

File: chapters/en/_toctree.yml [hunk 11]
```diff
@@ -3,7 +3,7 @@
   - local: chapter0/1
```
Comment (merveenoyan) [inline chapters/en/_toctree.yml:3]:
```suggestion
  - local: chapter0/introduction
```
and maybe we can rename the file too!

File: chapters/en/_toctree.yml [hunk 12]
```diff
@@ -3,7 +3,7 @@
...
     title: Introduction
 
-- title: Unit 1 - Fundamentals
+- title: Unit 1. Fundamentals
   sections:
   - sub-title: Chapter 1 - Fundamentals of Computer Vision
```
Comment (merveenoyan) [inline chapters/en/_toctree.yml:8]:
```suggestion
  - sub-title: Fundamentals of Computer Vision
```

File: chapters/en/_toctree.yml [hunk 13]
```diff
@@ -39,10 +82,80 @@
...
+      title: Introduction to Stable Diffusion
+  - sub-title: GANs & VAEs
+    sections:
+    - local: Unit 5 - Generative Models/GANs & VAEs/StyleGAN
+      title: StyleGAN Variants
+  - sub-title: Introduction
+    sections:
+    - local: Unit 5 - Generative Models/Introduction/Introduction
+      title: Introduction
+  - sub-title: PRACTICAL APPLICATIONS & CHALLENGES
```
Comment (merveenoyan) [inline chapters/en/_toctree.yml:102]:
```suggestion
  - sub-title: Practical Applications & Challenges
```

File: chapters/en/_toctree.yml [hunk 14]
```diff
@@ -39,10 +82,80 @@
...
+    sections:
+    - local: Unit 5 - Generative Models/PRACTICAL APPLICATIONS & CHALLENGES/Ethical Issues
```
Comment (merveenoyan) [inline chapters/en/_toctree.yml:104]:
```suggestion
    - local: Unit 5 - Generative Models/Practical Applications & Challenges/Ethical Issues
```
Can we change the folder name for consistency?

File: chapters/en/_toctree.yml [hunk 15]
```diff
@@ -39,10 +82,80 @@
...
+      title: Privacy, Bias and Societal Concerns
+    - local: Unit 5 - Generative Models/PRACTICAL APPLICATIONS & CHALLENGES/cycle_gan
```
Comment (merveenoyan) [inline chapters/en/_toctree.yml:106]:
```suggestion
    - local: Unit 5 - Generative Models/Practical Applications & Challenges/cycle_gan
```

File: chapters/en/_toctree.yml (hunk 5 above)
Comment (merveenoyan) [reply chapters/en/_toctree.yml:129]: I think there's an introduction chapter here: https://github.com/johko/computer-vision-course/tree/main/chapters/en/Unit%208%20-%203D%20Vision%2C%20Scene%20Rendering%20and%20Reconstruction/Introduction maybe we could look into introduction chapters first and remove this to be somewhere at the bottom

**Review by merveenoyan — COMMENTED** [review]
Thank you so much!!!

**Comment by ratan** [comment]
Thanks @merveenoyan, done the changes as suggested by you.
Please have a look

**Review by mmhamdy — COMMENTED** [review]
I have a suggestion, and I want to know what you think about it.

Why don't we just follow the folder naming structure used in [HF NLP course](https://github.com/huggingface/course/tree/main/chapters/en)? It would make the toc much simpler.

What do you think? @ratan @merveenoyan @bellabf

**Comment by ratan** [comment]
Thats a good suggestion. However the folder naming convention is is different. so toc file can be tuned for either of the way.
Lets @merveenoyan @bellabf also share what the final decision and the i will push the commit accordingly.

**Comment by bellabf** [comment]
> Thats a good suggestion. However the folder naming convention is is different. so toc file can be tuned for either of the way. Lets @merveenoyan @bellabf also share what the final decision and the i will push the commit accordingly.

Thank you for all your good work and quick reactivity! @ratan :smile: 

Great suggestion, @mmhamdy! I personally do not care too much on the style name the files as long as we follow the same style across all the course. IMO, the problem is that we are not aware of the order of the chapter for all the units (which I am guessing would make changing to the NLP convection harder, and there even be chapters that are self-contained and could fit in multiple places). The pro is that we would be forced to find out the order. The con is that if we get it wrong even once, there would be lots of renaming (which might make tracking the files a bit messy).

I think @merveenoyan has a better grasp on this and its long term impact, so I will wait for her opinion before re-reviewing the PR again.

Thanks again everyone :hugs:

**Review by mmhamdy — COMMENTED** [review]
> IMO, the problem is that we are not aware of the order of the chapter for all the units (which I am guessing would make changing to the NLP convection harder, and there even be chapters that are self-contained and could fit in multiple places). The pro is that we would be forced to find out the order.

Exactly @bellabf ! The chapters order problem is one we will have to face eventually, we just need to make sure we don't run into it again and get rid of different file naming styles.

It's just a little extra work, ordering and then asking authors to confirm it. What do you think @ratan ? I can also help with that if you need any help.

**Comment by ratan** [comment]
Hi @mmhamdy sure, you are most welcome. I have dropped DM to few authers but didnt get confirmation.
If you can help with ordering for 4, 5, 8, it will be really appreciated.

**Comment by mmhamdy** [comment]
Great! That sounds cool, I'll check the ordering in these units.

**Comment by merveenoyan** [comment]
@ratan we are following the one made for Audio Course: https://huggingface.co/learn/audio-course/chapter0/introduction

**Comment by kfahn22** [comment]
I just wanted to add that I think the READMEs in the Unit folders will need to either be moved (if you want to keep them for posterity) or deleted.

**Comment by ratan** [comment]
> @merveenoyan are you suggesting to change the chapter names as suggested inhttps://huggingface.co/learn/audio-course/chapter0/introduction in this repo and update toc file? Hope this doesn't create big rebase issue with other PR already being reviewed? We can divide it into 2 parts?
> 
> 1. merge this PR as such with order for 4,5,8 fixed
> 2. Later when 1 or 2 PR touching toc files is merged, i can create another PR just to change the chapter names
>    Please suggest @bellabf @mmhamdy

Please advise we if can go ahead with this suggestion?

**Comment by merveenoyan** [comment]
@ratan AFAIK it should've followed already, but we can open a new PR or handle it in this PR if possible

**Comment by merveenoyan** [comment]
@ratan if you can run styling in this PR we can merge it

**Review by merveenoyan — APPROVED** [review]

**Review by bellabf — APPROVED** [review]

**Comment by ratan** [comment]
merging the changes

**Pull request merged**